src/scan/grype_scan.go

```
@@ -0,0 +1,298 @@
+	if err != nil {
+		return "", fmt.Errorf("failed to create output file %s: %w", jsonOutputPath, err)
+	}
+	_ = file.Close()
+
+	args := []string{"sbom:" + sbomFile, "--add-cpes-if-none", "--output", "cyclonedx-json", "-v"}
+
+	args = append(args, "--file", jsonOutputPath)
+
+	// Try to scan with retries for database issues
+	return runGrypeCommand(args, jsonOutputPath, logger, isVerbose)
+}
+
+func scanImage(image, outputDir string, logger *slog.Logger, isVerbose bool) (string, error) {
+	logger.Debug("Scanning SBOM", slog.String("file", image))
+
+	// Set up the output path if needed
+	if outputDir == "" {
+		return "", errors.New("output directory not specified")
+	}
+
+	// If we couldn't extract the image ref, use the SBOM filename
+	safeImageName := sanitizeFilename(image)
+	jsonFileName := safeImageName + ".json"
+	jsonOutputPath := filepath.Join(outputDir, jsonFileName)
+	logger.Debug("Saving JSON results to output directory", slog.String("fileName", jsonOutputPath))
+
+	// Ensure the output directory exists and is writable
+	if err := os.MkdirAll(outputDir, 0755); err != nil {
+		return "", fmt.Errorf("failed to create output directory %s: %w", outputDir, err)
+	}
+
+	args := []string{image, "--add-cpes-if-none", "--output", "cyclonedx-json", "-v", "--file", jsonOutputPath}
+
+	// Try to scan with retries for database issues
+	return runGrypeCommand(args, jsonOutputPath, logger, isVerbose)
+}
+
+func runGrypeCommand(args []string, jsonOutputPath string, logger *slog.Logger, isVerbose bool) (string, error) {
+	// Maximum retry attempts for handling database issues
+	maxRetries := 3
+	retryCount := 0
+	for retryCount < maxRetries {
+		// Create the command - this needs to be inside the loop because we can't reuse commands
+
+		logger.Debug("Running grype command", slog.Any("args", args))
+		cmd := ExecCommand("grype", args...)
+		configureOutput(cmd, isVerbose)
+
+		// Print working directory and environment for debugging
+		wd, _ := os.Getwd()
```

Swallowing errors is never good idea :sweat: